compiler/rustc_hir_typeck/src/fn_ctxt/adjust_fulfillment_errors.rs

```diff
@@ -4,11 +4,26 @@ use rustc_hir::def::Res;
 use rustc_hir::def_id::DefId;
 use rustc_infer::traits::ObligationCauseCode;
 use rustc_middle::ty::{self, Ty, TyCtxt, TypeSuperVisitable, TypeVisitable, TypeVisitor};
-use rustc_span::{self, Span};
+use rustc_span::{self, symbol::kw, Span};
 use rustc_trait_selection::traits;
 
 use std::ops::ControlFlow;
 
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+enum ParamTerm {
```

Comment:
This should maybe live in `rustc_middle` next to `TermKind` since I could imagine us wanting to add a way to go from `TermKind` -> `ParamTerm`.